pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4784,39 +4782,61 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
-                if is_period is False and name in c_OFFSET_DEPR_FREQSTR:
+                if is_period is False and name.upper() in c_OFFSET_DEPR_FREQSTR:
+                    if n > 0 or stride == "-":
+                        name = name
```

Comment:
this looks very complex, and not sure it's necessary anyway?

Reply:
I agree, but I think it's the only way to pass `test_to_offset_invalid()` in `pandas/tests/tslibs/test_to_offset.py`
The reason is: when we parse frequency like `"2h20m"` or `"-m"` we show our warning `FutureWarning: 'm' is deprecated and will be removed in a future version, please use 'ME' instead.`
But here `'m'` is offsets.Hour(), not offsets.MonthEnd()

Reply:
sure but `'2h20m'` is already invalid to begin with?

I'm not sure we should worry about a false positive warning being raised for frequencies which were invalid to begin with